utils/monitor_soap.sh

```diff
@@ -0,0 +1,250 @@
+#!/bin/bash
+# 
+# This Script can trigger the soap run and monitor it until completion.
+# This script also takes care for editing the conf/global.properties file 
+# to set properties according to the current installation of zimbra
+# ######################################################################################
+function print_options {
+    cat << HELP_EOF
+	"$0" 
+        options:
+            --help			        Help
+            -u, --admin-user        Admin user name (Default: Admin)
+            -p, --admin-pass        Admin user passward (Required)
+            -h, --hostname          Hostname (optional)
+            -tp, --test-path        Tests path for tests to execute (optional)
+            -ts, --test-suite       Tests suite (optional)
+            -e, --notify-email      Email for sending test results. (Email will not be sent if no arg provided.)
+HELP_EOF
+}
+
+function main {
+    set -x
+    dir_bin="$(cd "$(dirname "$0")" && pwd)"
+    dir_qa="/opt/qa"
+    dir_staf="/usr/local/staf"
+    dir_soaprun="/opt/soaprun/"
+    notify_email="ganesh.anarse@syancor.com"
+    HOSTNAME="$(hostname -f)"
+    TEST_PATH="$dir_qa/soapvalidator/data/soapvalidator/Admin/Auth/AdminAuth-Basic.xml"
+    TEST_SUITE="BHR"
+    ADMIN_USER="admin"
+
+    while [ $# -gt 0 ]; do
+	key="$1"
+		case $key in
+			--help)
+			print_options
+			exit 0;
+			;;
+			-p|--admin-pass)
+			ADMIN_PASS=$2
+			shift
+			shift
+			;;
+			-u|--admin-user)
+			ADMIN_USER=$2
+			shift
+			shift
+			;;
+			-h|--hostname)
+			HOSTNAME=$2
+			shift
+			shift
+			;;
+			-tp|--test-path)
+			TEST_PATH=$2
+			shift
+			shift
+			;;
+			-ts|--test-suite)
+			TEST_SUITE=$2
+			shift
+			shift
+			;;
+			-e|--notify-email)
+			NOTIFY_EMAIL=$2
+			shift
+			shift
+			;;
+            *)
+			echo "Invalid argument(s): $2"
+			echo -e "\n\n\n"
+			print_options && exit 1
+			;;
+		esac
+	done
+
+    if [[ -z "$ADMIN_PASS" ]]; then
+        echo "--admin-pass is required" && exit 1;
+    fi
+    
+    check_staf
+    set_staf
+    set_properties
+    monitor_soap_run &
+}
+
+function monitor_soap_run {
+    soap_run_pid="$(run_tests)"
+    echo "Monitoring the SOAP run."
+    echo "PID: $soap_run_pid"
+    while kill -0 $soap_run_pid 2> /dev/null; do
+        sleep 30;
+    done
+    if [[ -n "${NOTIFY_EMAIL}" ]]; then
+        send_email_notification
+    fi
+}
+
+function send_email_notification {
+    echo "SOAP execution was exited. Emailing the results."
+    rm -rf ./email.txt
+    cat <<EOF > ./email.txt
+    To: $NOTIFY_EMAIL
+    Subject: "SOAP: Tests execution started at $(date)"
+
+    Status of the SOAP execution on $HOSTNAME
+    ------------------------------------------------------------
+    Host: $HOSTNAME
+    Test Path: $TEST_PATH
+    Test Suite: $TEST_SUITE
+    SOAP test results are now available at: $dir_soaprun
```

Comment:
can we mail the SOAP results as well ?

Reply:
Big file attachments do not get emailed correctly.
We will have to analyse the results with some function to consolidate it and email to the user.
I think we should take it up in some other ticket sir!